### ISSUE TITLE
Typing broken with @nuxtjs/i18n

### ISSUE THREAD

**Issue by ThomasBerne** [victorgarciaesgi/nuxt-typed-router#127]
**Describe the bug**
When adding the @nuxtjs/i18n module in a fresh project, the typing is broken:

![image](https://github.com/victorgarciaesgi/nuxt-typed-router/assets/23033483/7980a303-c52c-4b59-9d70-dc1819ce3b76)

**Expected behavior**
There should be typing generation in the install of the project and during the development using routing.

**Screenshots**

![image](https://github.com/victorgarciaesgi/nuxt-typed-router/assets/23033483/97b4c3a8-a889-4ce7-b547-ae4bdb077d83)

**Environnement infos**

Run `nuxi infos`

```
- Operating System: Linux
- Node Version:     v20.9.0
- Nuxt Version:     3.8.1
- CLI Version:      3.9.1
- Nitro Version:    2.7.2
- Package Manager:  pnpm@8.10.5
- Builder:          -
- User Config:      modules, i18n, security, tailwindcss, plugins, css, build, vite, devtools
- Runtime Modules:  @nuxt/devtools@1.0.2, nuxt-typed-router@3.3.3, @nuxtjs/i18n@8.0.0-rc.5, nuxt-security@1.0.0-rc.4, nuxt-simple-sitemap@4.1.3, @nuxtjs/robots@3.0.0, @vueuse/nuxt@10.6.1, @nuxtjs/tailwindcss@6.9.5, @pinia/nuxt@0.5.1, @nuxtjs/eslint-module@4.1.0, ()
- Build Modules:    -
```

Run `npx tree-node-cli {your page folder path}`

**Your nuxt.config.ts**
Nuxt i18n config : 
```ts
import type { NuxtI18nOptions } from '@nuxtjs/i18n/dist/module';
import { Locale } from '../../interfaces/locale.interface';

export const i18nConfig: NuxtI18nOptions = {
  vueI18n: './config/i18n/vue-i18n.ts',
  strategy: 'prefix',
  locales: [
    {
      code: Locale.EnUs,
      iso: Locale.EnUs,
      file: './config/i18n/locales/en-US.json',
    },
    {
      code: Locale.Frfr,
      iso: Locale.Frfr,
      file: './config/i18n/locales/fr-FR.json',
    },
  ],
};
```

And vue-i18n.ts content is:
```ts
export default defineI18nConfig(() => ({
  legacy: false,
  locale: 'en',
  numberFormats: {
    'en-US': {
      currency: {
        style: 'currency',
        currency: 'USD',
        notation: 'standard',
      },
    },
    'fr-FR': {
      currency: {
        style: 'currency',
        currency: 'EUR',
        notation: 'standard',
      },
    },
  },
}));
```

**Comment by ThomasBerne** [comment]
It looks like it's the vue-i18n.ts file that is breaking the typing. Removing the line : `vueI18n: './config/i18n/vue-i18n.ts',` in the module options fixes it.

**Comment by ThomasBerne** [comment]
i found the problem. I forgot this line:

![Uploading image.png…]()

**Comment by victorgarciaesgi** [comment]
Hi! So it's only the defaultLocale that caused the problem? I look at this

**Comment by victorgarciaesgi** [comment]
Fixed in `v3.5.0`
